File: src/main/java/com/solo/config/common/IdGenerator.java
package com.solo.config.common;

import java.util.UUID;

public class IdGenerator {

    private IdGenerator() {
    }

    public static String generate(String prefix) {
        return prefix + "_" + UUID.randomUUID().toString().replace("-", "").substring(0, 8);
    }

    public static String generateResourceId() {
        return generate("rsc");
    }

    public static String generateConfigId() {
        return generate("cfg");
    }

    public static String generateRunId() {
        return generate("run");
    }

    public static String generateSnapshotId() {
        return generate("snap");
    }

    public static String generateEventId() {
        return generate("evt");
    }

    public static String generateCommandId() {
        return generate("cmd");
    }

    public static String generateAuditId() {
        return generate("audit");
    }

    public static String generateNotificationId() {
        return generate("notif");
    }

    public static String generatePolicyId() {
        return generate("pol");
    }

    public static String generateInstanceId() {
        return generate("inst");
    }
}

File: src/utils/idGenerator.ts
import { v4 as uuidv4 } from 'uuid';

export const generateId = (prefix: string): string => {
  return `${prefix}_${uuidv4().replace(/-/g, '').substring(0, 12)}`;
};

export const generateRunId = (): string => generateId('run');
export const generateResourceId = (): string => generateId('rsc');
export const generateConfigId = (): string => generateId('cfg');
export const generateSnapshotId = (): string => generateId('snap');
export const generateCommandId = (): string => generateId('cmd');
export const generateAuditLogId = (): string => generateId('audit');
export const generateScenarioId = (): string => generateId('scn');
export const generateInjectionId = (): string => generateId('inj');
export const generatePolicyId = (): string => generateId('pol');
export const generateUpstreamId = (): string => generateId('ups');
export const generateCertId = (): string => generateId('cert');
export const generateRevocationId = (): string => generateId('rev');
export const generateLayerId = (): string => generateId('layer');
export const generateSyncTaskId = (): string => generateId('sync');
export const generateTemplateId = (): string => generateId('tpl');
export const generateEventId = (): string => generateId('evt');
